pkg/controller/machine/controller.go

```diff
@@ -95,6 +95,12 @@ func (c *MachineControllerImpl) Reconcile(machine *clusterv1.Machine) error {
 			return err
 		}
 
+		// Fetch a fresh Machine object.
+		machine, err := c.Get(machine.Namespace, name)
```

Comment:
this might not be a fresh copy, as Get uses the lister cache.
if you really do want to use the cache, you should make a DeepCopy of it, as it is a read-only, local cache.

Reply:
Ok I've updated it to use the client itself.